label_studio/data_manager/managers.py

```diff
@@ -498,8 +513,56 @@ def __init__(self, expression, distinct=False, output_field=None, **extra):
 def annotate_completed_at(queryset):
     from tasks.models import Annotation
 
+    LseProject = load_func(settings.LSE_PROJECT)
+    get_tasks_agreement_queryset = load_func(settings.GET_TASKS_AGREEMENT_QUERYSET)
+
     newest = Annotation.objects.filter(task=OuterRef('pk')).order_by('-id')[:1]
-    return queryset.annotate(completed_at=Case(When(is_labeled=True, then=Subquery(newest.values('created_at')))))
+    if (
+        get_tasks_agreement_queryset
+        and LseProject
+        and queryset
+        and LseProject.objects.filter(project_id=queryset[0].project_id).exists()
+    ):
+        queryset = get_tasks_agreement_queryset(queryset)
+
+        # Subquery to get the agreement_threshold for each project
+        agreement_threshold_subquery = Subquery(
+            LseProject.objects.filter(project_id=OuterRef('project_id')).values('agreement_threshold')[:1],
+            output_field=FloatField(),
+        )
```

Comment:
OuterRef('project_id') is need it or is always the same project id. 
Also coming from this
`LseProject.objects.filter(project_id=queryset[0].project_id).exists(`
if we check only the first i guess all are the same.
Sorry from throw comment without read the complete file